cli/diff.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+use crate::colors;
+use dissimilar::{diff, Chunk};
+use std::path::PathBuf;
+
+macro_rules! print_add {
+  () => {
+    print!("{}", colors::green_bold("+".to_string()));
+  };
+}
+
+macro_rules! fmt_add_text {
+  ($x:expr) => {
+    format!("{}", colors::green($x));
+  };
+}
+
+macro_rules! print_rem {
+  () => {
+    print!("{}", colors::red_bold("-".to_string()));
+  };
+}
+
+macro_rules! fmt_rem_text {
+  ($x:expr) => {
+    format!("{}", colors::red($x));
+  };
+}
+
+macro_rules! fmt_add_text_highlight {
+  ($x:expr) => {
+    format!("{}", colors::white_on_green($x));
+  };
+}
+
+macro_rules! fmt_rem_text_highlight {
+  ($x:expr) => {
+    format!("{}", colors::white_on_red($x));
+  };
+}
+
+fn print_line_diff(
+  orig_line: &mut usize,
+  edit_line: &mut usize,
+  line_number_width: usize,
+  orig: &mut String,
+  edit: &mut String,
+) {
+  let print_line_number = |line: usize| {
+    print!(
+      "{:0width$}{} ",
+      line,
+      colors::gray("|".to_string()),
+      width = line_number_width
+    );
+  };
+
+  let split = orig.split('\n').enumerate();
+  for (i, s) in split {
+    print_line_number(*orig_line + i);
+    print_rem!();
+    print!("{}", s);
+    println!();
+  }
+
+  let split = edit.split('\n').enumerate();
+  for (i, s) in split {
+    print_line_number(*edit_line + i);
+    print_add!();
+    print!("{}", s);
+    println!();
+  }
+
+  *orig_line += orig.split('\n').count();
+  *edit_line += edit.split('\n').count();
+
+  orig.clear();
+  edit.clear();
+}
+
+/// Print diff of the same file_path, before and after formatting.
+///
+/// Diff format is loosely based on Github diff formatting.
+pub fn print_diff(file_path: &PathBuf, orig_text: &str, edit_text: &str) {
```

Comment:
Rather than having this function print to stdout directly, it might be better to have it return a string, so that we can add a test here.

Reply:
Good call